controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -221,23 +224,50 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 
 	// check request status already contains the name of the resource. if not, add it.
 	if r.StorageClassRequest.Spec.Type == "blockpool" {
+		// initialize in-memory structs
+		r.cephRadosNamespace = &rookCephv1.CephBlockPoolRadosNamespace{}
+		r.cephRadosNamespace.Namespace = r.OperatorNamespace
 		r.cephBlockPool = &rookCephv1.CephBlockPool{}
 		r.cephBlockPool.Namespace = r.OperatorNamespace
-		for _, res := range r.StorageClassRequest.Status.CephResources {
-			if res.Kind == "CephBlockPool" {
-				r.cephBlockPool.Name = res.Name
-				break
-			}
+
+		// check if a CephBlockPoolRadosNamespace resource exists for the desired storageconsumer and storageprofile.
+		cephRadosNamespaceList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+		err := r.Client.List(
+			r.ctx,
+			cephRadosNamespaceList,
+			client.InNamespace(r.OperatorNamespace),
+			client.MatchingFields{util.OwnerUIDIndexName: string(r.StorageClassRequest.UID)})
+		if err != nil {
+			return err
+		}
+
+		// if we found no CephBlockPoolRadosNamespaces, generate a new name
+		// if we found only one CephBlockPoolRadosNamespace with our query, we're good
+		// if we found more than one CephBlockPoolRadosNamespace, we can't determine which one to select, so error out
+		rnsItemsLen := len(cephRadosNamespaceList.Items)
+		if rnsItemsLen == 0 {
+			rnsNewName := fmt.Sprintf("cephradosnamespace-%s-%s", r.StorageClassRequest.Spec.StorageProfile, r.storageConsumer.Name)
+			r.log.V(1).Info("no valid CephBlockPoolRadosNamespace found, creating new one", "CephBlockPoolRadosNamespace", rnsNewName)
+			r.cephRadosNamespace.Name = rnsNewName
+		} else if rnsItemsLen == 1 {
+			cephRns := cephRadosNamespaceList.Items[0]
+			r.cephBlockPool.Name = cephRns.Spec.BlockPoolName
+			r.cephRadosNamespace = &cephRns
+			r.log.V(1).Info("valid CephBlockPoolRadosNamespace found", "CephBlockPoolRadosNamespace", r.cephRadosNamespace.Name)
+		} else {
+			return fmt.Errorf("invalid number of CephBlockPoolRadosNamespaces for storage consumer %q and storage profile %q: found %d, expecting 0 or 1", r.storageConsumer.Name, profileName, rnsItemsLen)
 		}
 
-		// check if a cephblockpool resource exists for the desired storageconsumer and storageprofile.
+		// check if a cephblockpool resource exists for the desired storageprofile.
 		if r.cephBlockPool.Name == "" {
 			cephBlockPoolList := &rookCephv1.CephBlockPoolList{}
-			listOptions := &client.MatchingLabels{
-				controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
-				controllers.StorageProfileSpecLabel:  storageProfile.GetSpecHash(),
+			listOptions := []client.ListOption{
+				client.InNamespace(r.OperatorNamespace),
+				&client.MatchingLabels{
+					controllers.StorageProfileLabel: profileName,
```

Comment:
What if the profile name is empty and we need to resort to the default blockpool/ FS datapool?

Right now the profile name is identified by the following code
```
	profileName := r.StorageClassRequest.Spec.StorageProfile
	if profileName == "" {
		profileName = r.storageCluster.Spec.DefaultStorageProfile
	}
```

Which after the design change doesn't make sense as we are deprecating StorageProfle usage and the `DefaultStorageProfile` field

Reply:
The default block pool doesn't not have a fixed name, it is generated from the StorageCluster name. And if this is a legitimate provider-mode deployment there shouldn't be any default block pool at all. Per your requests elsewhere, if the StorageClassRequest has no specified profile and the StorageCluster has no default profile, we will fail.

Reply:
@jarrpa 
> The default block pool doesn't have a fixed name, it is generated from the StorageCluster name

Even if it does not have a fixed name it has a predictable name that we can use. 

> And if this is a legitimate provider-mode deployment 

There is no such a thing anymore, the point is to converge in implementation and on storage administration behavior

> Per your requests elsewhere, if the StorageClassRequest has no specified profile and the StorageCluster has no default profile, we will fail.

After these changes `default profile = default pool`, we will always have a default pool so if a profile name was not given you should be able to fulfill the request 100% of the time. The other comment mentioned a case when a user specify an explicit storage profile name, but we don't have a pool with that name.  

On a more general note:
Block pools are capturing storage concerns/requirements (which are defined by the env and the user). Namespaces are capturing access requirements (which are defined by the provider's multi-tenancy model). The entire point of using namespaces and moving away from creating new pools on behalf of the user it to give back the responsibilities of storage administration to the user while keeping access administration inside ODF (Provider mode) automation.